File: java/aws/lambda/src/main/java/example/MockLambdaLogger.java
package example;

import com.amazonaws.services.lambda.runtime.LambdaLogger;

public class MockLambdaLogger implements LambdaLogger {

    @Override
    public void log(String message) {
        System.out.println("[LOG] " + message);
    }

    @Override
    public void log(byte[] message) {
        System.out.println("[LOG] " + new String(message));
    }
}

File: java/aws/lambda-kotlin/src/test/kotlin/example/invoke/MockLambdaLogger.kt
package example.invoke

import com.amazonaws.services.lambda.runtime.LambdaLogger
import com.amazonaws.services.lambda.runtime.logging.LogLevel

class MockLambdaLogger : LambdaLogger {

    override fun log(message: String) {
        println("[LOG] $message")
    }

    override fun log(message: String, logLevel: LogLevel) {
        println("[$logLevel] $message")
    }

    override fun log(message: ByteArray) {
        println("[LOG] ${String(message)}")
    }

    override fun log(message: ByteArray, logLevel: LogLevel) {
        println("[$logLevel] ${String(message)}")
    }
}
